src/Series.cpp

```diff
@@ -85,145 +97,105 @@ Series::Series(std::string filepath,
         filepath = auxiliary::replace_all(filepath, "\\", "/");
     }
 #endif
-    std::string newPath;
-    std::string newName;
     auto const pos = filepath.find_last_of(auxiliary::directory_separator);
     if( std::string::npos == pos )
     {
-        newPath = ".";
-        newPath.append(1, auxiliary::directory_separator);
-        newName = filepath;
+        input->path = ".";
+        input->path.append(1, auxiliary::directory_separator);
+        input->name = filepath;
     }
     else
     {
-        newPath = filepath.substr(0, pos + 1);
-        newName = filepath.substr(pos + 1);
+        input->path = filepath.substr(0, pos + 1);
+        input->name = filepath.substr(pos + 1);
     }
 
-    IterationEncoding ie;
-    if( std::string::npos != newName.find("%T") )
-        ie = IterationEncoding::fileBased;
+    input->format = determineFormat(input->name);
+
+#if defined(IS_GCC_48)
+    regex_t pattern;
+    if( regcomp(&pattern, "(.*)%(0\\d+)?T(.*)", REG_EXTENDED) )
+        throw std::runtime_error(std::string("Regex for iterationFormat '(.*)%(0\\d+)?T(.*)' can not be compiled!"));
+
+    uint8_t maxGroups = 4;
+    regmatch_t regexMatch[maxGroups];
+
+    if( regexec(&pattern, input->name.c_str(), maxGroups, regexMatch, 0) )
+        input->iterationEncoding = IterationEncoding::groupBased;
     else
-        ie = IterationEncoding::groupBased;
+    {
+        input->iterationEncoding = IterationEncoding::fileBased;
 
-    m_format = std::make_shared< Format >(determineFormat(newName));
+        if( regexMatch[0].rm_so == -1 || regexMatch[0].rm_eo == -1 )
+            throw std::runtime_error("Can not determine iterationFormat from filename " + input->name);
 
-    m_writable->IOHandler = AbstractIOHandler::createIOHandler(newPath, at, *m_format, comm);
-    IOHandler = m_writable->IOHandler.get();
-    iterations.linkHierarchy(m_writable);
+        if( regexMatch[1].rm_so == -1 || regexMatch[1].rm_eo == -1 )
+            throw std::runtime_error("Can not determine iterationFormat (prefix) from filename " + input->name);
+        else
+            input->filenamePrefix = std::string(&input->name[regexMatch[1].rm_so], regexMatch[1].rm_eo);
 
-    m_name = std::make_shared< std::string >(cleanFilename(newName, *m_format));
+        if( regexMatch[2].rm_so == -1 || regexMatch[2].rm_eo == -1 )
+            input->filenamePadding = 0;
+        else
+            input->filenamePadding = std::stoi(std::string(&input->name[regexMatch[2].rm_so], regexMatch[2].rm_eo));
 
-    switch( at )
+        if( regexMatch[3].rm_so == -1 || regexMatch[3].rm_eo == -1 )
+            throw std::runtime_error("Can not determine iterationFormat (postfix) from filename " + input->name);
+        else
+            input->filenamePrefix = std::string(&input->name[regexMatch[3].rm_so], regexMatch[3].rm_eo);
+    }
+
+    regfree(&pattern);
+#else
+    std::regex pattern("(.*)%(0\\d+)?T(.*)");
```

Comment:
@ax3l this is a PoC of https://github.com/openPMD/openPMD-api/issues/265#issuecomment-397604092